graphql-cli/src/main/java/io/ballerina/graphql/cmd/GraphqlCmd.java

```diff
@@ -0,0 +1,374 @@
+            description = "Directory to store the generated Ballerina clients. " +
+                    "If this is not provided, the generated files will be stored in the current execution directory.")
+    private String outputPath;
+
+    @CommandLine.Parameters
+    private List<String> argList;
+
+    /**
+     * Constructor that initialize with the default values.
+     */
+    public GraphqlCmd() {
+        this.outStream = System.err;
+        this.exitWhenFinish = true;
+    }
+
+    /**
+     * Constructor override, which takes output stream and execution dir as inputs.
+     *
+     * @param outStream      output stream from ballerina
+     * @param executionDir   defines the directory location of  execution of ballerina command
+     */
+    public GraphqlCmd(PrintStream outStream, Path executionDir) {
+        new GraphqlCmd(outStream, executionDir, true);
+    }
+
+    /**
+     * Constructor override, which takes output stream and execution dir and exits when finish as inputs.
+     *
+     * @param outStream         output stream from ballerina
+     * @param executionDir      defines the directory location of  execution of ballerina command
+     * @param exitWhenFinish    exit when finish the execution
+     */
+    public GraphqlCmd(PrintStream outStream, Path executionDir, boolean exitWhenFinish) {
+        this.outStream = outStream;
+        this.executionPath = executionDir;
+        this.exitWhenFinish = exitWhenFinish;
+    }
+
+    @Override
+    public void execute() {
+
+        // Check if CLI help flag argument is present
+        if (helpFlag) {
+            // TODO: Send a PR with cli-help/ballerina-graphql.help file to
+            //  https://github.com/ballerina-platform/ballerina-lang/tree/master/cli/ballerina-cli/src/main/resources/
+//            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(getName());
+//            outStream.println(commandUsageInfo);
+            return;
+        }
+
+        // Check if CLI input argument is present
+        if (inputPath) {
+            // Check if GraphQL configuration file is provided
+            if (argList == null) {
+                outStream.println(MESSAGE_FOR_MISSING_INPUT_ARGUMENT);
+                exitError(this.exitWhenFinish);
+                return;
+            }
+
+            String fileName = argList.get(0);
+            if (fileName.endsWith(YAML_EXTENSION)) {
+                try {
+                    CodeGenerator codeGenerator = new CodeGenerator();
+
+                    GraphqlConfig graphqlConfig = getGraphQLConfig(fileName);
+                    validateGraphQLConfig(graphqlConfig);
+
+                    if (graphqlConfig.getProjects() != null) {
+                        for (String projectName : graphqlConfig.getProjects().keySet()) {
+
+                            validateExistenceOfProject(graphqlConfig.getProjects().get(projectName));
+                            validateExistenceOfSchemaAndDocuments(
```

Comment:
This should come under `validateExistenceOfProject` - shouldn't it?

Reply:
Yes. This is moved to `validateExistenceOfProject` function now & changes are amended.